runtime/commonTest/src/kotlinx/serialization/features/PolymorphismTest.kt

```diff
@@ -27,22 +27,54 @@ class PolymorphismTest {
     private val json = Json { unquotedPrint = true; useArrayPolymorphism = true; serialModule = module }
 
     @Test
-    fun testInheritanceJson() {
+    fun testInheritanceJson() = parametrizedTest { useStreaming ->
         val obj = Wrapper(
             PolyBase(2),
             PolyDerived("b")
         )
-        val bytes = json.stringify(Wrapper.serializer(), obj)
+        val bytes = json.stringify(Wrapper.serializer(), obj, useStreaming)
         assertEquals(
             "{polyBase1:[kotlinx.serialization.PolyBase,{id:2}]," +
                     "polyBase2:[kotlinx.serialization.PolyDerived,{id:1,s:b}]}", bytes
         )
     }
 
     @Test
-    fun testSerializeWithExplicitPolymorhpicSerializer() {
+    fun testSerializeWithExplicitPolymorphicSerializer() = parametrizedTest { useStreaming ->
         val obj = PolyDerived("b")
-        val s = json.stringify(PolymorphicSerializer(PolyDerived::class), obj)
+        val s = json.stringify(PolymorphicSerializer(PolyDerived::class), obj, useStreaming)
         assertEquals("[kotlinx.serialization.PolyDerived,{id:1,s:b}]", s)
     }
+
+    object PolyDefaultSerializer : JsonTransformingSerializer<PolyDefault>(PolyDefault.serializer(), "foo") {
+        override fun readTransform(element: JsonElement): JsonElement {
+            return json {
+                "json" to element
+                "id" to 42
```

Comment:
This one is not really **that** convenient, especially the fact that I should specify `PolyDefault.serializer()`.
Maybe we can improve it even more